.github/workflows/prover-test.yml

```diff
@@ -53,3 +64,14 @@ jobs:
         run: |
           cd light-prover
           go test -timeout 30m
+
+      - name: Extract circuit to Lean
+        run: |
+          cd light-prover
+          ./light-prover extract-circuit --output formal-verification/FormalVerification/Circuit.lean --tree-depth=20 --compressed-accounts=10
```

Comment:
This command extracts the definition from golang into the `circuit.lean` file – that is it makes sure we actually work based on your implementation and not our idea of it. `Main.lean` imports that file and proves the theorems. It will stop compiling if `Circuit.lean` changes (which will happen if the circuit definition in golang changes).

Depth and number of inputs have no impact on any of this – I've changed it to 26/8 for you and the only changes I've had to make was find-and-replace the autogenerated circuit names.